### PR TITLE
Advanced search

### PR DESCRIPTION
**Pull request by gabeholley222**
Working grade, degree, and projSize attributes. Still need to implement the actual searchability.

### CODE REVIEW

File: src/ypd/servlets/project_servlet.py [hunk 1]
```diff
@@ -74,10 +74,17 @@ def archive(self, project):
...
+        if request.method == 'POST':
                 projType = form.projType.data
                 title = form.title.data
                 description = form.description.data
+                electrical = DegreeAttributes.electrical.value in form.degree.data
+                mechanical = DegreeAttributes.mechanical.value in form.degree.data
+                computer = DegreeAttributes.computer.value in form.degree.data
+                computersci = DegreeAttributes.computersci.value in form.degree.data
+                grade = GradeAttributes(form.grade.data)
+                print(f'Grade Value: {grade}')
```
Comment (caydenreynolds) [inline src/ypd/servlets/project_servlet.py:86]: We should remove this debugging print, my guy

File: src/ypd/templates/project.html [hunk 2]
```diff
@@ -30,7 +30,7 @@ <h1>{{project.title}}</h1>
         </a>
     </div>
 
-    {% if project.can_be_modified_by(user) %}
+    {% if project.can_be_modified_by(user) and project.archived == False %}
```
Comment (caydenreynolds) [inline src/ypd/templates/project.html:33]: Why are we protecting archived projects from modification?

File: src/ypd/form/project_form.py [hunk 3]
```diff
@@ -1,18 +1,34 @@
...
 
-    title = StringField('Title:', validators=[InputRequired()])
+    
+
+    title = StringField('Title:', validators=[InputRequired()], widget=TextArea(), render_kw={'cols': '150', 'rows': '2'})
     description = TextAreaField('Project Summary:', validators=[InputRequired()],
                                 widget=TextArea(), render_kw={'cols': '150', 'rows': '25'})
     projType = RadioField('Project Type:', validators=[InputRequired()], coerce=int,
                           choices=[(PROVIDED, 'Provided Project'), (SOLICITED,'Solicited Project')])
+    maxProjSize = IntegerField("Maximum Number of People Needed: ")
```
Comment (caydenreynolds) [inline src/ypd/form/project_form.py:19]: If I leave the MaxProjSize box blank, It gets set to a null value in the database. We should flash an error to the user instead

Comment (gabeholley222) [reply src/ypd/form/project_form.py:19]: Set it so that if left blank, sets maxProjSize to 1

File: src/ypd/servlets/project_servlet.py (hunk 1 above)
Comment (gabeholley222) [reply src/ypd/servlets/project_servlet.py]: Removed

File: src/ypd/templates/project.html (hunk 2 above)
Comment (gabeholley222) [reply src/ypd/templates/project.html:33]: The almighty gods have spoken

File: src/ypd/form/project_form.py (hunk 3 above)
Comment (gabeholley222) [reply src/ypd/form/project_form.py:19]: Flashes error if no entry

**Pull request merged**